### DISCUSSION TITLE
What is the best practice to announce a security issue in my project?

### DISCUSSION THREAD

**Discussion by paijp** [community/community#51445] [New to GitHub] [answered]
-- I found and fixed a command injection security issue in my project.
-- I believe no users are using this project.
-- I wrote this problem in the 'important changes' section of README.md ( https://github.com/paijp/nophp ).
-- What is the best practice to announce this?

**Comment by ramazancetinkaya** [comment, marked as answer]
When it comes to announcing a security issue in your project on GitHub, there are some best practices that you should follow to ensure that the information is communicated clearly and effectively to your users. Here are some steps you can take:

1) Notify the relevant parties first: Before making any public announcement, it's important to notify any affected parties first, such as users of your software or service, any downstream dependencies, or any companies or organizations that may be impacted. You should provide them with as much detail as possible about the issue and any steps they should take to protect themselves.

2) Create a security advisory: GitHub provides a dedicated security advisory feature that allows you to create a public record of the security issue, its impact, and any remediation steps. This is a good place to communicate all relevant details in a clear and concise way. You can also use this feature to track the progress of the issue, including any fixes or patches that have been released.

3) Provide a timeline: Users will want to know when they can expect a fix or patch for the issue. Providing a timeline for when you expect to release a fix can help manage expectations and reduce confusion.

4) Be transparent: Transparency is key when it comes to security issues. You should provide as much information as possible about the issue, including how it was discovered, its impact, and any steps that have been taken to remediate it. This can help build trust with your users and demonstrate that you take security seriously.

5) Communicate clearly: Make sure that your communication about the security issue is clear and easy to understand. Avoid using technical jargon or overly complex language that may confuse users.

6) Encourage users to take action: Finally, you should encourage your users to take any necessary action to protect themselves from the security issue. This may include updating their software, changing passwords, or taking other steps to secure their accounts or systems.

---

Hope this clarifies.

If you find this reply useful, please kindly 'Mark as answer' so that others know this issue is already addressed.

Have a good day.

**Reply by paijp** [reply]
-- Thank you very much!
-- I wrote a security advisory. https://github.com/paijp/nophp/security/advisories/GHSA-9858-q3c2-9wwm
-- This project is a framework. The user may be an engineer. But I think I have to write the details of "command injection" .

**Reply by paijp** [reply]
-- I was saved a little by the message "Thank you for making the open source ecosystem more secure by fixing and responsibly disclosing this vulnerability." .
